torch/csrc/dynamo/extra_state.cpp

```diff
@@ -86,9 +87,15 @@ PyObject* lookup(ExtraState* extra_state, PyObject* f_locals) {
   CacheEntry* found = nullptr;
   py::handle locals(f_locals);
   for (CacheEntry& cache_entry : extra_state->cache_entry_list) {
-    py::object valid = py::none();
+    bool valid = false;
     try {
-      valid = cache_entry.check_fn(locals);
+      // TODO(anijain2305) - Clean this up when enable_cpp_guard_manager is True
+      // by default
+      if (cache_entry.root_mgr != nullptr) {
+        valid = run_root_guard_manager(cache_entry.root_mgr, f_locals);
```

Comment:
Note, I suspect there is a bit of extra perf to be had by not generating f_locals and instead using fastlocals directly.

Reply:
Yes, directly working on fastlocals is on the table for guard refactor.